### DISCUSSION TITLE
Pinning of pylint

### DISCUSSION THREAD

**Discussion by espenfl** [aiidateam/aiida-core#6053] [General]
There is now this https://github.com/aiidateam/aiida-core/blob/d0e1de261881daec87c452369f17c1f2533e3fe5/pyproject.toml#L108 in `aiida-core`. This is causing issues at the plugin side where we sometimes need to use a more updated `pylint`, more specifically `pylint~=2.15` due to the issues with `wrapt` and Python 3.11.

Can we do something with the current pinning?

**Comment by mbercx** [comment]
> Can we do something with the current pinning?

I see no reason not to switch to the looser `pylint~=2.11` specification. What do you think, dependency manager @sphuber? ^^

**Reply by sphuber** [reply]
The reason we pin to the minor version is because they release new features with new minor versions that will break the pre-commit, which is annoying. That being said, there is nothing against making a PR that updates the minor version, and would in fact be very welcome. In the end though, the `pylint` dependency is only specified in the `pre-commit` optional dependencies, so why is this a problem for plugins? Why are you installed `aiida-core[pre-commit]` in your plugin? You can just install plain `aiida-core` and then define your own pre-commit dependencies, right @espenfl ?

**Reply by espenfl** [reply]
Great. When developing in both AiiDA and for plugins it is nice to use the extras from `aiida-core`. At least for `aiida-vasp` we have moved to this (still waiting for a no-install functionality though for `pre-commit`). We could revert it, but since there is an issue in `wrapt` that has been fixed for Python 3.11 (maybe you do not hit it in the `aiida-core` tests) I think it is worthwhile to consider to even bump it to `~=2.15`. Seems the fixed `wrapt` is included from there onwards. Happy to adjust on the `aiida-vasp` side if this is too much work.

**Reply by sphuber** [reply]
It's going to be a lot of work updating to 2.17 as that will come with a shit ton of new warnings to correct. So not sure when that will be done. If you are looking for a quick fix, I think you might be better of trying to fix it by getting rid of relying on AiiDA's pre-commit dependency

**Reply by espenfl** [reply]
I know as we did it on our side so I totally understand it. Also, some of the usual suspects have been moved, which at least for our case makes the configs and overrides less noisy. Then we will do with our own setup for the `pre-commit` for now. Thanks.

**Reply by ltalirz** [reply]
Just mentioning that there is also the option to upgrade pylint, but add any new rules to the ignore list (and opening an issue to reintroduce them later).

And, finally, for plugins you can opt to install pylint in its own environment like the other hooks (i.e. not using `language: system`) - the only downside is that you will need to also install all your other dependencies there since they are required for the linting but I have seen it before

**Reply by espenfl** [reply]
> And, finally, for plugins you can opt to install pylint in its own environment like the other hooks (i.e. not using language: system) - the only downside is that you will need to also install all your other dependencies there since they are required for the linting but I have seen it before

That is indeed a very possible route to follow. Thanks for suggesting that. The main goal for us was to not bother with the `tests` and `pre-commit` environments and just accept whatever is done in `aiida-core`, since the plugin is tied to it anyway.

**Reply by sphuber** [reply]
https://github.com/aiidateam/aiida-core/pull/6054
